crates/rooch/src/commands/move_cli/commands/run_function.rs

```diff
@@ -153,52 +153,28 @@ impl CommandAction<ExecuteTransactionResponseView> for RunFunction {
                 context.execute(tx).await?
             }
             (None, None) => {
-                if context.keystore.get_if_password_is_empty() {
-                    context
-                        .sign_and_execute(sender, action, None, max_gas_amount)
-                        .await?
-                } else {
-                    let password =
-                        prompt_password("Enter the password to run functions:").unwrap_or_default();
-                    let is_verified = verify_password(
-                        Some(password.clone()),
-                        context.keystore.get_password_hash(),
+                let tx_data = context
+                    .build_tx_data(sender, action.clone(), max_gas_amount)
+                    .await?;
+                let tx_execution_result = context.sign_and_execute(sender, tx_data.clone()).await?;
+
+                if self.gas_profile {
+                    //TODO FIXME we should use the state_root from previous tx
+                    let state_root = tx_execution_result
+                        .execution_info
+                        .state_root
+                        .0
+                        .as_bytes()
+                        .to_vec();
```

Comment:
@steelgeek091 We use a wrong state_root to execute_tx locally.

Reply:
We getting the last Tx order which has processed by sequencer then use this Tx order to query previous Tx?

Reply:
If the tx has been executed and you use the state_root to execute it again, the ObjectID and Resource may conflict. Maybe the gas profile mode should only execute the tx off-chain.